src/lib.rs

```diff
@@ -861,26 +908,35 @@ impl fmt::Debug for Client {
     }
 }
 
-struct Key {
+/// This struct represents a key used to sign requests to the ACME server
+pub struct Key {
     rng: crypto::SystemRandom,
     signing_algorithm: SigningAlgorithm,
     inner: crypto::EcdsaKeyPair,
     thumb: String,
 }
 
 impl Key {
-    fn generate() -> Result<(Self, crypto::pkcs8::Document), Error> {
+    /// Generate a random key
+    pub fn generate() -> Result<(Self, crypto::pkcs8::Document), Error> {
         let rng = crypto::SystemRandom::new();
         let pkcs8 =
             crypto::EcdsaKeyPair::generate_pkcs8(&crypto::ECDSA_P256_SHA256_FIXED_SIGNING, &rng)?;
         Self::new(pkcs8.as_ref(), rng).map(|key| (key, pkcs8))
     }
 
-    fn from_pkcs8_der(pkcs8_der: &[u8]) -> Result<Self, Error> {
+    /// Create a key from a der encoded ES256
+    pub fn from_pkcs8_der(pkcs8_der: &[u8]) -> Result<Self, Error> {
         Self::new(pkcs8_der, crypto::SystemRandom::new())
     }
 
-    fn new(pkcs8_der: &[u8], rng: crypto::SystemRandom) -> Result<Self, Error> {
+    /// Export a key to a der encoded ES256 key
+    pub fn to_pkcs8_der(&self) -> Result<crypto::pkcs8::Document, Error> {
+        Ok(self.inner.to_pkcs8v1()?)
+    }
+
+    /// Create a key from a der encoded ES256 key and a crypto::SystemRandom
```

Comment:
If this fn stays I think similar feedback as the `from_pkcs8_der()` specificity applies.